src/GameLogic.cpp

```diff
@@ -146,13 +150,38 @@ void GameLogic::updateProjectileCollisions() {
     //does nothing yet, eventually loop through projectiles and 
     //check if they are hitting the player or an enemy
     for(int p = 0; p < projectiles.size(); ++p) {
-            for(int e = 0; e < enemies.size(); ++e) {
-                if (projectiles[p].get().getSprite().getGlobalBounds().intersects( enemies[e].get().getSprite().getGlobalBounds())) {
-                    enemies[e].get().setOffScreen();
+        if (projectiles[p].get().getType() == 0) {
+            if (projectiles[p].get().getSprite().getGlobalBounds().intersects(player.getSprite().getGlobalBounds())) {
+                softReset();
+                projectiles[p].get().setAvailable();
+                removeFromActors(projectiles[p].get());
+                projectiles.erase(projectiles.begin() + p);
+
+            }
+        }
+        else { //must be bullet, type is 1
```

Comment:
I get that the type can only be 1 if it's not 0, but you should still be explicit to avoid unintended errors down the line. Just make it an else if checking the type.